From 5c7c41688d6f28235a5cabe3e275266e9bcb958e Mon Sep 17 00:00:00 2001
From: polirritmico <ebray187@gmail.com>
Date: Wed, 9 Aug 2023 17:52:29 -0400
Subject: [PATCH] Default File View folder to QStandard music path

The default folder could be changed from the System settings:
    System settings -> Applications -> Locations -> Music path
---
 src/viewslistdata.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/viewslistdata.cpp b/src/viewslistdata.cpp
index 13da83ec..c4176fa7 100644
--- a/src/viewslistdata.cpp
+++ b/src/viewslistdata.cpp
@@ -16,6 +16,7 @@
 
 #include <QUrl>
 #include <QDir>
+#include <QStandardPaths>
 
 class ViewsListDataPrivate
 {
@@ -130,7 +131,11 @@ public:
                                                ViewManager::DelegateWithoutSecondaryText,
                                                ViewManager::ViewHideRating,
                                                ViewManager::IsFlatModel,
-                                               QUrl::fromLocalFile(QDir::rootPath())
+                                               QUrl::fromLocalFile(
+                                                       !QStandardPaths::writableLocation(QStandardPaths::MusicLocation).isEmpty() and QDir(QStandardPaths::writableLocation(QStandardPaths::MusicLocation)).exists() ?
+                                                       QStandardPaths::writableLocation(QStandardPaths::MusicLocation) :
+                                                       QDir::rootPath()
+                                                       )
                                               },
                                               {{i18nc("@title:window Title of the file radios browser view", "Radio Stations")},
                                                QUrl{QStringLiteral("image://icon/radio")},
-- 
2.41.0

